deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,244 @@
+                                                                    self._epsilon).reset_index()
+
+        return metric_results, examples
+
+    def _get_random_image_pairs_from_dataloder(self, baseline_data_loader, augmented_data_loader, n_samples=10):
+        """
+        We iterate the internal dataset object directly to avoid randomness
+        Dataset returns data points as processed images, making this currently not really usable
+        To avoid making more assumptions this currently stays as-is
+        Note that images return in RGB format, ond to visualize them using OpenCV the final dimesion should be
+        transposed;
+        can be done via image = image[:, :, ::-1] or cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
+        """
+        # We definitely make the assumption that the underlying structure is torch.utils.Dataset
+        baseline_sampler = iter(baseline_data_loader.dataset)
+        aug_sampler = iter(augmented_data_loader.dataset)
+        samples = []
+        # iterate and sample
+        for idx, (sample_base, sample_aug) in enumerate(zip(baseline_sampler, aug_sampler)):
+            if idx > n_samples:
+                break
+            samples.append((sample_base[0], sample_aug[0]))
+
+        return samples
+
+    def _create_example_figure(self, example_dict):
+        # TODO not fully implemented
+        from plotly.subplots import make_subplots
+        import plotly.graph_objects as go
+        figures = []
+        for aug_name, aug_images in example_dict.items():
+            fig = make_subplots(rows=3, cols=3)
+            for i in range(3):
+                for j in range(3):
+                    print(j + (i * 3))
+                    fig.add_trace(
+                        go.Image(z=aug_images[j + (i * 3)]), row=i + 1, col=j + 1
+                    )
+            fig.update_layout(height=600, width=800)
+            figures.append(fig)
+        return figures
+
+    def _set_seeds(self, seed):
+        """
+        Sets seeds for reproduceability
+        Imgaug uses numpy's State
+        Albumentation uses Python and imgaug seeds
+        :param seed:
+        :return:
+        """
+        np.random.seed(seed)
+        random.seed(seed)
+        torch.manual_seed(seed)
+        if torch.cuda.is_available():
+            torch.cuda.manual_seed_all(seed)
+
+
+def get_robustness_augmentations(transform_type):
+    if transform_type == 'albumentations':
+        # Note that p=1.0 since we want to apply those to entire dataset
+        return [
+            albumentations.RandomBrightnessContrast(p=1.0),
+            albumentations.ShiftScaleRotate(p=1.0),
+            albumentations.HueSaturationValue(p=1.0),
+            albumentations.RGBShift(r_shift_limit=15, g_shift_limit=15, b_shift_limit=15, p=1.0),
+        ]
+    # imgaug augmentations works also inside pytorch compose
+    else:
+        return [
+            imgaug.augmenters.MultiplyHueAndSaturation()
```

Comment:
Need to expand to match the albumentations version if possible, but can do in a later PR.